### PR TITLE
feat: truncate table in standalone mode

### PR DESCRIPTION
**Pull request by DevilExileSu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
1.  Implement truncate table procedure in standalone mode.
2.  Implement reset version. 
3.  Implement truncate region.
    1.  Acquires the `RegionWriter` lock.
    2.  Create `RegionMetaAction::Truncate` to store `committed_sequence` and persist it for recovery from the manifest.
    3.  Mark all data in the WAL as `obsolete`.
    4.  Mark all SSTs as deleted.
    5.  Reset Version. 
4.  Recover from `RegionMetaAction::Truncate`.
    1.  Mark all SSTs as deleted.
    2.  Reset the Version and set the `committed_sequence` from `RegionMetaAction::Truncate` as the Version's `flushed_sequence`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datanode/src/sql/truncate_table.rs [hunk 1]
```diff
@@ -28,8 +32,29 @@ impl SqlHandler {
...
+        let engine_procedure = self.engine_procedure(table)?;
+
+        let procedure =
+            TruncateTableProcedure::new(req, self.catalog_manager.clone(), engine_procedure);
+
+        let procedure_with_id = ProcedureWithId::with_random_id(Box::new(procedure));
+        let procedure_id = procedure_with_id.id;
+
+        info!(
+            "Truncate table {} by procedure {}",
```
Comment (evenyag) [inline src/datanode/src/sql/truncate_table.rs:45]: nitpick(non-blocking): Also print table id?

File: src/table-procedure/src/truncate.rs [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+#[cfg(test)]
+mod tests {
+    use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+    use table::engine::TableEngine;
+
+    use super::*;
+    use crate::test_util::TestEnv;
+
+    #[tokio::test]
+    async fn test_drop_table_procedure() {
```
Comment (evenyag) [inline src/table-procedure/src/truncate.rs:242]:
```suggestion
    async fn test_truncate_table_procedure() {
```

File: src/datanode/src/sql/truncate_table.rs [hunk 3]
```diff
@@ -28,8 +32,32 @@ impl SqlHandler {
...
+            req.clone(),
+            self.catalog_manager.clone(),
+            engine_procedure,
+        );
+
+        let procedure_with_id = ProcedureWithId::with_random_id(Box::new(procedure));
+        let procedure_id = procedure_with_id.id;
+
+        info!(
+            "Truncate table {}, table-id {} by procedure {}",
```
Comment (evenyag) [inline src/datanode/src/sql/truncate_table.rs:48]:
```suggestion
            "Truncate table {}, table_id {} by procedure {}",
```

**Review by evenyag — APPROVED** [review]
Rest LGTM

**Review by WenyXu — APPROVED** [review]

**Comment by evenyag** [comment]
@DevilExileSu Could you please format the code?

**Pull request merged**